classes/01class/exercises/c01-git04/alanlima/c01-git04.md

```diff
@@ -0,0 +1,43 @@
+# Exercise C01-GIT04
+
+1. Let's suppose you remotely pushed your my_env.txt file. A colleague asks you to remove this information from Git. What's your colleague worried about?
+
+    ***This file exposes sensitive information (password) which will be visible for anyone with access to the repository.***
+
+2. If you modify the file in your workspace, then push it, will it be enough? (It's not). Why?
+
+    ***Nope, someone can check the `git log` and checkout the file version with the sensitive data and read its contents.***
+
+3. If you delete the file and push it, then create a new one with the rest of the information, is it enough? (It's not). Why?
+
+    ***Nope, you delete the file in the recent commit but not from the history. Someone can check the `git log` and checkout the file's commit and read its contents.***
+
+4. How to fix this? How do you remove something from Git history?
+
+    ***We need to execute the same procedures as when we need to remove a big file form the git history,
+    we need to rewrite it and push the new history to the remote repository.
+    We can use either `BFG Repo-Cleaner` or `git filter-branch` to completely remove the file from git history.***
```

Comment:
Remember that any person with the commit SHA can view a commit (even after removed from the tree) using the UI, e.g., https://github.com/devopsacademyau/academy/commit/<COMMIT_SHA>

For definitely removing a commit, a new repo would be necessary.